### PR TITLE
ARROW-10746: [C++] Bump gtest version + use GTEST_SKIP in tests

### PR DESCRIPTION
**Pull request by arw2019**
As per a TODO left in ARROW-3769 / #3721 we can now use the `GTEST_SKIP` macro in `parquet/encoding-test.cpp`. `GTEST_SKIP` was added in gtest 1.10.0 so this involves bumping our minimal gtest version from 1.8.1

### CODE REVIEW

**Comment by arw2019** [comment]
I think the code changes here are ok. However, at least some of the build errors look related (and persisted across two CI runs) so that's left to figure out

**Comment by pitrou** [comment]
Well, at least the ASAN CI job failure seems related:
```
clang: error: no such file or directory: 'debug//libgtestd.so'
```

cc @kou @xhochy

**Comment by kou** [comment]
It seems that the latest GoogleTest doesn't use the default `CMAKE_XXX_OUTPUT_DIRECTORY`: https://github.com/google/googletest/blob/master/googletest/cmake/internal_utils.cmake#L159-L164

Could you try this?

```diff
diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
index df03c3129..e36571808 100644
--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
@@ -1603,21 +1603,16 @@ macro(build_gtest)
   set(GTEST_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/googletest_ep-prefix")
   set(GTEST_INCLUDE_DIR "${GTEST_PREFIX}/include")
 
-  set(_GTEST_RUNTIME_DIR ${BUILD_OUTPUT_ROOT_DIRECTORY})
+  set(_GTEST_LIBRARY_DIR "${GTEST_PREFIX}/lib")
 
   if(MSVC)
     set(_GTEST_IMPORTED_TYPE IMPORTED_IMPLIB)
     set(_GTEST_LIBRARY_SUFFIX
         "${CMAKE_GTEST_DEBUG_EXTENSION}${CMAKE_IMPORT_LIBRARY_SUFFIX}")
-    # Use the import libraries from the EP
-    set(_GTEST_LIBRARY_DIR "${GTEST_PREFIX}/lib")
   else()
     set(_GTEST_IMPORTED_TYPE IMPORTED_LOCATION)
     set(_GTEST_LIBRARY_SUFFIX
         "${CMAKE_GTEST_DEBUG_EXTENSION}${CMAKE_SHARED_LIBRARY_SUFFIX}")
-
-    # Library and runtime same on non-Windows
-    set(_GTEST_LIBRARY_DIR "${_GTEST_RUNTIME_DIR}")
   endif()
 
   set(GTEST_SHARED_LIB
@@ -1632,6 +1627,7 @@ macro(build_gtest)
       ${EP_COMMON_TOOLCHAIN}
       -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
       "-DCMAKE_INSTALL_PREFIX=${GTEST_PREFIX}"
+      -DCMAKE_INSTALL_LIBDIR=lib
       -DBUILD_SHARED_LIBS=ON
       -DCMAKE_CXX_FLAGS=${GTEST_CMAKE_CXX_FLAGS}
       -DCMAKE_CXX_FLAGS_${UPPERCASE_BUILD_TYPE}=${GTEST_CMAKE_CXX_FLAGS})
@@ -1641,27 +1637,6 @@ macro(build_gtest)
     set(GTEST_CMAKE_ARGS ${GTEST_CMAKE_ARGS} "-DCMAKE_MACOSX_RPATH:BOOL=ON")
   endif()
 
-  if(CMAKE_GENERATOR STREQUAL "Xcode")
-    # Xcode projects support multi-configuration builds.  This forces the gtest build
-    # to use the same output directory as a single-configuration Makefile driven build.
-    list(
-      APPEND GTEST_CMAKE_ARGS "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=${_GTEST_LIBRARY_DIR}"
-             "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY_${CMAKE_BUILD_TYPE}=${_GTEST_RUNTIME_DIR}")
-  endif()
-
-  if(MSVC)
-    if(NOT ("${CMAKE_GENERATOR}" STREQUAL "Ninja"))
-      set(_GTEST_RUNTIME_DIR ${_GTEST_RUNTIME_DIR}/${CMAKE_BUILD_TYPE})
-    endif()
-    set(GTEST_CMAKE_ARGS
-        ${GTEST_CMAKE_ARGS} "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY=${_GTEST_RUNTIME_DIR}"
-        "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY_${CMAKE_BUILD_TYPE}=${_GTEST_RUNTIME_DIR}")
-  else()
-    list(
-      APPEND GTEST_CMAKE_ARGS "-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=${_GTEST_RUNTIME_DIR}"
-             "-DCMAKE_RUNTIME_OUTPUT_DIRECTORY_${CMAKE_BUILD_TYPE}=${_GTEST_RUNTIME_DIR}")
-  endif()
-
   add_definitions(-DGTEST_LINKED_AS_SHARED_LIBRARY=1)
 
   if(MSVC AND NOT ARROW_USE_STATIC_CRT)
```

**Comment by kou** [comment]
It seems that the template type name `T` is conflicted with Visual C++. Could you try this patch?

```diff
diff --git a/cpp/src/parquet/statistics_test.cc b/cpp/src/parquet/statistics_test.cc
index 0828f36b3..77e77e0eb 100644
--- a/cpp/src/parquet/statistics_test.cc
+++ b/cpp/src/parquet/statistics_test.cc
@@ -609,7 +609,7 @@ static const int NUM_VALUES = 10;
 template <typename TestType>
 class TestStatisticsSortOrder : public ::testing::Test {
  public:
-  typedef typename TestType::c_type T;
+  using c_type = typename TestType::c_type;
 
   void AddNodes(std::string name) {
     fields_.push_back(schema::PrimitiveNode::Make(
@@ -670,7 +670,7 @@ class TestStatisticsSortOrder : public ::testing::Test {
   }
 
  protected:
-  std::vector<T> values_;
+  std::vector<c_type> values_;
   std::vector<uint8_t> values_buf_;
   std::vector<schema::NodePtr> fields_;
   std::shared_ptr<schema::GroupNode> schema_;
@@ -700,13 +700,13 @@ void TestStatisticsSortOrder<Int32Type>::SetValues() {
 
   // Write UINT32 min/max values
   stats_[0]
-      .set_min(std::string(reinterpret_cast<const char*>(&values_[5]), sizeof(T)))
-      .set_max(std::string(reinterpret_cast<const char*>(&values_[4]), sizeof(T)));
+      .set_min(std::string(reinterpret_cast<const char*>(&values_[5]), sizeof(c_type)))
+      .set_max(std::string(reinterpret_cast<const char*>(&values_[4]), sizeof(c_type)));
 
   // Write INT32 min/max values
   stats_[1]
-      .set_min(std::string(reinterpret_cast<const char*>(&values_[0]), sizeof(T)))
-      .set_max(std::string(reinterpret_cast<const char*>(&values_[9]), sizeof(T)));
+      .set_min(std::string(reinterpret_cast<const char*>(&values_[0]), sizeof(c_type)))
+      .set_max(std::string(reinterpret_cast<const char*>(&values_[9]), sizeof(c_type)));
 }
 
 // TYPE::INT64
@@ -728,13 +728,13 @@ void TestStatisticsSortOrder<Int64Type>::SetValues() {
 
   // Write UINT64 min/max values
   stats_[0]
-      .set_min(std::string(reinterpret_cast<const char*>(&values_[5]), sizeof(T)))
-      .set_max(std::string(reinterpret_cast<const char*>(&values_[4]), sizeof(T)));
+      .set_min(std::string(reinterpret_cast<const char*>(&values_[5]), sizeof(c_type)))
+      .set_max(std::string(reinterpret_cast<const char*>(&values_[4]), sizeof(c_type)));
 
   // Write INT64 min/max values
   stats_[1]
-      .set_min(std::string(reinterpret_cast<const char*>(&values_[0]), sizeof(T)))
-      .set_max(std::string(reinterpret_cast<const char*>(&values_[9]), sizeof(T)));
+      .set_min(std::string(reinterpret_cast<const char*>(&values_[0]), sizeof(c_type)))
+      .set_max(std::string(reinterpret_cast<const char*>(&values_[9]), sizeof(c_type)));
 }
 
 // TYPE::FLOAT
@@ -747,8 +747,8 @@ void TestStatisticsSortOrder<FloatType>::SetValues() {
 
   // Write Float min/max values
   stats_[0]
-      .set_min(std::string(reinterpret_cast<const char*>(&values_[0]), sizeof(T)))
-      .set_max(std::string(reinterpret_cast<const char*>(&values_[9]), sizeof(T)));
+      .set_min(std::string(reinterpret_cast<const char*>(&values_[0]), sizeof(c_type)))
+      .set_max(std::string(reinterpret_cast<const char*>(&values_[9]), sizeof(c_type)));
 }
 
 // TYPE::DOUBLE
@@ -761,8 +761,8 @@ void TestStatisticsSortOrder<DoubleType>::SetValues() {
 
   // Write Double min/max values
   stats_[0]
-      .set_min(std::string(reinterpret_cast<const char*>(&values_[0]), sizeof(T)))
-      .set_max(std::string(reinterpret_cast<const char*>(&values_[9]), sizeof(T)));
+      .set_min(std::string(reinterpret_cast<const char*>(&values_[0]), sizeof(c_type)))
+      .set_max(std::string(reinterpret_cast<const char*>(&values_[9]), sizeof(c_type)));
 }
 
 // TYPE::ByteArray
```

Here is a patch to fix CMake lint failure:

```diff
diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
index bdf15e4bc..46872fba3 100644
--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
@@ -1603,7 +1603,7 @@ macro(build_gtest)
   set(GTEST_PREFIX "${CMAKE_CURRENT_BINARY_DIR}/googletest_ep-prefix")
   set(GTEST_INCLUDE_DIR "${GTEST_PREFIX}/include")
 
-set(_GTEST_LIBRARY_DIR "${GTEST_PREFIX}/lib")
+  set(_GTEST_LIBRARY_DIR "${GTEST_PREFIX}/lib")
 
   if(MSVC)
     set(_GTEST_IMPORTED_TYPE IMPORTED_IMPLIB)
```

**Comment by kou** [comment]
Could you add required GoogleTest version check?

```diff
diff --git a/cpp/cmake_modules/ThirdpartyToolchain.cmake b/cpp/cmake_modules/ThirdpartyToolchain.cmake
index bdf15e4bc..60bdf13ad 100644
--- a/cpp/cmake_modules/ThirdpartyToolchain.cmake
+++ b/cpp/cmake_modules/ThirdpartyToolchain.cmake
@@ -1682,7 +1682,7 @@ set(_GTEST_LIBRARY_DIR "${GTEST_PREFIX}/lib")
 endmacro()
 
 if(ARROW_TESTING)
-  resolve_dependency(GTest)
+  resolve_dependency(GTest REQUIRED_VERSION 1.10.0)
 
   if(NOT GTEST_VENDORED)
     # TODO(wesm): This logic does not work correctly with the MSVC static libraries
```

**Comment by arw2019** [comment]
Thanks @kou!!! Pushed your suggestions

**Comment by arw2019** [comment]
Is this related in C++ / AMD64 Windows 2019? 
```
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(527,33): error C2039: 'TearDownTestCase': is not a member of 'parquet::ByteArray' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-writer-test.vcxproj]
D:\a\arrow\arrow\cpp\src\parquet/types.h(518): message : see declaration of 'parquet::ByteArray' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-writer-test.vcxproj]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(525): message : while compiling class template member function 'testing::internal::SetUpTearDownSuiteFuncType testing::internal::SuiteApiResolver<parquet::test::TestByteArrayValuesWriter>::GetTearDownCaseOrSuite(const char *,int)' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-writer-test.vcxproj]
D:/a/arrow/arrow/cpp/src/parquet/column_writer_test.cc(688): message : see reference to function template instantiation 'testing::internal::SetUpTearDownSuiteFuncType testing::internal::SuiteApiResolver<parquet::test::TestByteArrayValuesWriter>::GetTearDownCaseOrSuite(const char *,int)' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-writer-test.vcxproj]
D:/a/arrow/arrow/cpp/src/parquet/column_writer_test.cc(688): message : see reference to class template instantiation 'testing::internal::SuiteApiResolver<parquet::test::TestByteArrayValuesWriter>' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-writer-test.vcxproj]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(529,33): error C2039: 'TearDownTestSuite': is not a member of 'parquet::ByteArray' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-writer-test.vcxproj]
D:\a\arrow\arrow\cpp\src\parquet/types.h(518): message : see declaration of 'parquet::ByteArray' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-writer-test.vcxproj]
```

**Comment by kou** [comment]
I think so.
We need a fix like https://github.com/apache/arrow/pull/8782#issuecomment-736217882 for `cpp/src/parquet/column_writer_test.cc`.

File: cpp/src/parquet/encoding_test.cc [hunk 1]
```diff
@@ -495,7 +488,9 @@ class TestArrowBuilderDecoding : public ::testing::Test {
   void CheckDecodeArrowNonNullUsingDenseBuilder() {
     for (auto np : null_probabilities_) {
       InitTestCase(np);
-      SKIP_TEST_IF(null_count_ > 0)
+      if (null_count_ > 0) {
+        GTEST_SKIP();
```
Comment (pitrou) [inline cpp/src/parquet/encoding_test.cc:492]: This should probably be `continue` instead.

File: cpp/src/parquet/encoding_test.cc [hunk 2]
```diff
@@ -508,7 +503,9 @@ class TestArrowBuilderDecoding : public ::testing::Test {
   void CheckDecodeArrowNonNullUsingDictBuilder() {
     for (auto np : null_probabilities_) {
       InitTestCase(np);
-      SKIP_TEST_IF(null_count_ > 0)
+      if (null_count_ > 0) {
+        GTEST_SKIP();
```
Comment (pitrou) [inline cpp/src/parquet/encoding_test.cc:507]: Same here.

**Review by pitrou — CHANGES_REQUESTED** [review]
Just two small comments.

File: cpp/src/parquet/encoding_test.cc (hunk 1 above)
Comment (arw2019) [reply cpp/src/parquet/encoding_test.cc]: Right, done

File: cpp/src/parquet/encoding_test.cc (hunk 2 above)
Comment (arw2019) [reply cpp/src/parquet/encoding_test.cc]: Done

**Comment by arw2019** [comment]
It's down to three failing checks:
- Appveyor - not sure what the problem is
- C++ / AMD64 MacOS 10.15 - looks like a `gtest` location problem
```
dyld: Library not loaded: @rpath/libgtest_maind.dylib
```
- C++/ AMD64 Windows 2019 - a template instantiation problem in `cpp/src/parquet/statistics_test.cc`

```
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(527,33): error C2039: 'TearDownTestCase': is not a member of 'parquet::ByteArray' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
D:\a\arrow\arrow\cpp\src\parquet/types.h(518): message : see declaration of 'parquet::ByteArray' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(525): message : while compiling class template member function 'testing::internal::SetUpTearDownSuiteFuncType testing::internal::SuiteApiResolver<TestClass>::GetTearDownCaseOrSuite(const char *,int)' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(705): message : see reference to function template instantiation 'testing::internal::SetUpTearDownSuiteFuncType testing::internal::SuiteApiResolver<TestClass>::GetTearDownCaseOrSuite(const char *,int)' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(706): message : see reference to class template instantiation 'testing::internal::SuiteApiResolver<TestClass>' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(698): message : while compiling class template member function 'bool testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types3<T2,T3,T4>>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::ByteArrayType,
              T3=parquet::FLBAType,
              T4=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(720): message : see reference to function template instantiation 'bool testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types3<T2,T3,T4>>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::ByteArrayType,
              T3=parquet::FLBAType,
              T4=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(721): message : see reference to class template instantiation 'testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types3<T2,T3,T4>>' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::ByteArrayType,
              T3=parquet::FLBAType,
              T4=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(698): message : while compiling class template member function 'bool testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types4<T2,T3,T4,T5>>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::DoubleType,
              T3=parquet::ByteArrayType,
              T4=parquet::FLBAType,
              T5=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(720): message : see reference to function template instantiation 'bool testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types4<T2,T3,T4,T5>>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::DoubleType,
              T3=parquet::ByteArrayType,
              T4=parquet::FLBAType,
              T5=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(721): message : see reference to class template instantiation 'testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types4<T2,T3,T4,T5>>' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::DoubleType,
              T3=parquet::ByteArrayType,
              T4=parquet::FLBAType,
              T5=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(698): message : while compiling class template member function 'bool testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types5<T2,T3,T4,T5,T6>>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::FloatType,
              T3=parquet::DoubleType,
              T4=parquet::ByteArrayType,
              T5=parquet::FLBAType,
              T6=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(720): message : see reference to function template instantiation 'bool testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types5<T2,T3,T4,T5,T6>>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::FloatType,
              T3=parquet::DoubleType,
              T4=parquet::ByteArrayType,
              T5=parquet::FLBAType,
              T6=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(721): message : see reference to class template instantiation 'testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types5<T2,T3,T4,T5,T6>>' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::FloatType,
              T3=parquet::DoubleType,
              T4=parquet::ByteArrayType,
              T5=parquet::FLBAType,
              T6=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(698): message : while compiling class template member function 'bool testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types6<T2,T3,T4,T5,T6,T7>>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::Int64Type,
              T3=parquet::FloatType,
              T4=parquet::DoubleType,
              T5=parquet::ByteArrayType,
              T6=parquet::FLBAType,
              T7=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(720): message : see reference to function template instantiation 'bool testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types6<T2,T3,T4,T5,T6,T7>>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::Int64Type,
              T3=parquet::FloatType,
              T4=parquet::DoubleType,
              T5=parquet::ByteArrayType,
              T6=parquet::FLBAType,
              T7=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(721): message : see reference to class template instantiation 'testing::internal::TypeParameterizedTest<Fixture,TestSel,testing::internal::Types6<T2,T3,T4,T5,T6,T7>>' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
          with
          [
              Fixture=parquet::test::TestStatistics,
              TestSel=testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,
              T2=parquet::Int64Type,
              T3=parquet::FloatType,
              T4=parquet::DoubleType,
              T5=parquet::ByteArrayType,
              T6=parquet::FLBAType,
              T7=parquet::BooleanType
          ]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(698): message : while compiling class template member function 'bool testing::internal::TypeParameterizedTest<parquet::test::TestStatistics,testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,parquet::test::gtest_type_params_TestStatistics_>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
D:/a/arrow/arrow/cpp/src/parquet/statistics_test.cc(490): message : see reference to function template instantiation 'bool testing::internal::TypeParameterizedTest<parquet::test::TestStatistics,testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,parquet::test::gtest_type_params_TestStatistics_>::Register(const char *,const testing::internal::CodeLocation &,const char *,const char *,int,const std::vector<std::string,std::allocator<std::string>> &)' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
D:/a/arrow/arrow/cpp/src/parquet/statistics_test.cc(490): message : see reference to class template instantiation 'testing::internal::TypeParameterizedTest<parquet::test::TestStatistics,testing::internal::TemplateSel<parquet::test::TestStatistics_MinMaxEncode_Test>,parquet::test::gtest_type_params_TestStatistics_>' being compiled [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
D:\a\arrow\arrow\build\cpp\googletest_ep-prefix\include\gtest/internal/gtest-internal.h(529,33): error C2039: 'TearDownTestSuite': is not a member of 'parquet::ByteArray' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
D:\a\arrow\arrow\cpp\src\parquet/types.h(518): message : see declaration of 'parquet::ByteArray' [D:\a\arrow\arrow\build\cpp\src\parquet\parquet-internals-test.vcxproj]
```
I'll keep looking.

**Comment by kou** [comment]
Both the Appveyor case and the macOS case are caused by same reason. They can't find GoogleTest's library.
I'll take a look them.

`cpp/src/parquet/statistics_test.cc`: It seems that naming conflict like other build failures but I'm not sure. I'll also take a look it.

**Comment by kou** [comment]
@github-actions crossbow submit -g nightly

**Review by kou — APPROVED** [review]
+1

All green. I'll merge this.

**Comment by arw2019** [comment]
Thanks @kou @pitrou !!!

**Pull request closed without merging**